### PR TITLE
Suppot linux-arm config

### PR DESCRIPTION
**Pull request by sergey-nbl**

### CODE REVIEW

File: src/perfEvent.cpp [hunk 1]
```diff
@@ -25,7 +25,14 @@
 #include "perfEvent.h"
 #include "vmEntry.h"
 
+// from aarch32_port
+#if (defined(__ARM_ARCH) && __ARM_ARCH >= 7) || defined(__ARM_ARCH_7A__) || defined(__ARM_ARCH_7__)
+#define rmb() __asm__ __volatile__ ("dmb ish"   : : : "memory")
+#elif defined (__arm__) || (__thumb__)
```
Comment (apangin) [inline src/perfEvent.cpp]: `(__thumb__)` --> `defined(__thumb__)` ?

File: src/spinLock.h [hunk 2]
```diff
@@ -17,8 +17,12 @@
 #ifndef _SPINLOCK_H
 #define _SPINLOCK_H
 
+#if defined(__arm__) || defined(__thumb__)
+#include <sched.h>
+#define pause() sched_yield()
+#else
 #define x86_pause()  asm volatile("pause")
```
Comment (apangin) [inline src/spinLock.h]: In this case `x86_pause()` should be also renamed. But `pause()` is a library function defined in <unistd.h>

File: src/spinLock.h [hunk 3]
```diff
@@ -17,8 +17,12 @@
 #ifndef _SPINLOCK_H
 #define _SPINLOCK_H
 
+#if defined(__arm__) || defined(__thumb__)
+#include <sched.h>
+#define pause() sched_yield()
```
Comment (apangin) [inline src/spinLock.h]: ARMv6K+ has `YIELD` instruction that is similar to x86 `PAUSE`

File: src/stackFrame.cpp [hunk 4]
```diff
@@ -17,7 +17,19 @@
 #include <ucontext.h>
 #include "stackFrame.h"
 
+#if defined(__arm__) || defined(__thimb__)
```
Comment (apangin) [inline src/stackFrame.cpp]: typo: thumb

**Comment by sergey-nbl** [comment]
Ouch! So many error, will fix later without haste.

And yes, it works on RPi2

**Comment by sergey-nbl** [comment]
OK, I've updated the files. Now I limited a number of supported CPU by ARMv7 and higher due to DMB and YIELD instructions need. Verified on RPI2, build check only for x86

**Pull request closed without merging**